### PR TITLE
Update base image to RH UBI 9 and Golang 1.21.13

### PR DESCRIPTION
**Pull request by vkmc**

### CODE REVIEW

**Review by jlarriba — APPROVED** [review]
/lgtm

**Comment by vyzigold** [comment]
I think we don't have yet everything setup correctly for PRs to merge automatically. This lgtm too and the CI is green, so I'll merge it.

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Integration testing
 env:
   PROJECT_ROOT: /root/go/src/github.com/infrawatch/sg-core
-  OPSTOOLS_REPO: https://raw.githubusercontent.com/infrawatch/sg-core/04dcb34edd2c234b378222d2f9a17e15c0dad936/build/repos/opstools.repo
+  OPENSTACK_REPO: https://trunk.rdoproject.org/centos9-caracal/delorean-deps.repo
```
Comment (compi-migui) [inline .github/workflows/integration.yml:4]: question (non-blocking): This repo file provides 6 repositories on top of the `centos-opstools` we already had. Do we ever expect to install RPMs from those new ones?

Looking at the image build and integration runs it seems we end up taking just qpid-proton-c from `centos9-opstools` and everything else comes in from UBI repos.

**Review by compi-migui — APPROVED** [review]

File: .github/workflows/integration.yml (hunk 1 above)
Comment (vkmc) [reply .github/workflows/integration.yml:4]: Good catch Miguel. We don't expect to use new packages. I couldn't find qpid-proton-c in the opstools repo, that's the reason I enabled OpenStack ones. Maybe there is a lighter repo we can use?

Comment (compi-migui) [reply .github/workflows/integration.yml:4]: Seems to be in opstools: https://mirror.stream.centos.org/SIGs/9-stream/opstools/
Namely https://mirror.stream.centos.org/SIGs/9-stream/opstools/x86_64/collectd-5/Packages/q/qpid-proton-c-0.39.0-2.el9s.x86_64.rpm

If that's what we need and we want to minimize repos we enable, we could keep just centos9-opstools from https://github.com/openstack-k8s-operators/sg-core/blob/main/build/repos/centos9-caracal.repo. Maybe rename the file back to opstools.repo.

Like I said this is all non-blocking. Just some tidying up.

**Pull request merged**